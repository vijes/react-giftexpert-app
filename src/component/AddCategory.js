import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {

    const [inputvalue, setinputvalue] = useState('')

    const handleInputChange = ( e ) => {
        // console.log(e.target.value);
        setinputvalue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputvalue ) {
            setCategories( cats => [ inputvalue, ...cats ]);
            setinputvalue('');
        }else{
            console.log('No debe ser vacio el input');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p> { inputvalue } </p>
            <input 
                type="text"
                value={ inputvalue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}