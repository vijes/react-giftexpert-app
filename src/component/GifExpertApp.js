import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['One punch', 'Samurai X', 'Dragon Ball'];

    // const [categories, setCategories] = useState(['Samurai X']);
    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     setCategories( ['Dragon Ball GT', ...categories] );
    //     // setCategories( cats => [ ... cats, 'Dagon Ball GT']);
    // }

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;