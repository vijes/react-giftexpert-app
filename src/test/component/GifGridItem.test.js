import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../component/GifGridItem';


describe('Pruebas de GifGridItem', () => {
    
    const title = 'Titulo test';
    const url = 'https://localhost:3000/icon.jpg/';
    let wrapper = shallow(<GifGridItem title={ title } url={ url } />);

    beforeEach( () => {
        wrapper = shallow(<GifGridItem title={ title } url={ url } />);
    });

    test('debe de mostrar el componente correctamente', () => {
        // const wrapper = shallow(<GifGridItem />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
    });

    test('debe de obtener la imagen igual al url y alt de los props.', () => {

        const img = wrapper.find('img');
        // console.log( img.props() );
        // console.log( img.prop('src') );
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
        
    });
    
    test('debe tener el class name animate__swing', () => {
        
        const div = wrapper.find('div');
        const className = div.prop( 'className' );

        expect( className.includes( 'animate__swing' ) ).toBe( true );
    })
        
    
})
