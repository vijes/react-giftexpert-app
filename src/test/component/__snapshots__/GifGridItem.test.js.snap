// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__swing"
>
  <img
    alt="Titulo test"
    src="https://localhost:3000/icon.jpg/"
  />
  <p>
     
    Titulo test
     
  </p>
</div>
`;
